engine/apps/mattermost/tests/test_tasks.py

```diff
@@ -0,0 +1,214 @@
+import json
+
+import httpretty
+import pytest
+from django.conf import settings
+from django.utils import timezone
+from rest_framework import status
+
+from apps.alerts.models import AlertGroupLogRecord
+from apps.mattermost.client import MattermostAPIException
+from apps.mattermost.models import MattermostMessage
+from apps.mattermost.tasks import on_alert_group_action_triggered_async, on_create_alert_async
+
+
+@pytest.mark.django_db
+@httpretty.activate(verbose=True, allow_net_connect=False)
+def test_on_create_alert_async_success(
+    make_organization_and_user,
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_mattermost_post_response,
+):
+    organization, _ = make_organization_and_user()
+    alert_receive_channel = make_alert_receive_channel(organization)
+    alert_group = make_alert_group(alert_receive_channel)
+    alert = make_alert(alert_group=alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+
+    url = "{}/api/v4/posts".format(settings.MATTERMOST_HOST)
+    data = make_mattermost_post_response()
+    mock_response = httpretty.Response(json.dumps(data), status=status.HTTP_200_OK)
+    httpretty.register_uri(httpretty.POST, url, responses=[mock_response])
+
+    on_create_alert_async(alert_pk=alert.pk)
+
+    mattermost_message = alert_group.mattermost_messages.order_by("created_at").first()
+    assert mattermost_message.post_id == data["id"]
+    assert mattermost_message.channel_id == data["channel_id"]
+    assert mattermost_message.message_type == MattermostMessage.ALERT_GROUP_MESSAGE
+
+
+@pytest.mark.django_db
+@httpretty.activate(verbose=True, allow_net_connect=False)
+def test_on_create_alert_async_skip_post_for_duplicate(
+    make_organization_and_user,
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_mattermost_post_response,
+    make_mattermost_message,
+):
+    organization, _ = make_organization_and_user()
+    alert_receive_channel = make_alert_receive_channel(organization)
+    alert_group = make_alert_group(alert_receive_channel)
+    alert = make_alert(alert_group=alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+    make_mattermost_message(alert_group, MattermostMessage.ALERT_GROUP_MESSAGE)
+
+    url = "{}/api/v4/posts".format(settings.MATTERMOST_HOST)
+    data = make_mattermost_post_response()
+    mock_response = httpretty.Response(json.dumps(data), status=status.HTTP_200_OK)
+    httpretty.register_uri(httpretty.POST, url, responses=[mock_response])
+
+    on_create_alert_async(alert_pk=alert.pk)
+
+    request = httpretty.last_request()
+    assert request.url is None
```

Comment:
Curious (didn't check this before), is this the way to check there was no request made to the patched URL?